### PR TITLE
Added Kmeans algorithm using C

### LINKED ISSUE #1795: Implementation of K-means clustering algorithm using C
Would like to add the K-means clustering algorithm implemented in C.
@asha15 @singh-shreya6 @jainaman224 @raghav-dalmia please assign me this

### PR DESCRIPTION
**Pull request by arghac14**
Fixes #1795 

#### Checklist:
- [X] 4 space indentation.
- [X] Coding conventions are followed.
- [X] Input is taken dynamically.
- [X] Sample Input / Output is added at the end of file.
- [X] Logic Documentation (Comments).
- [X] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Added Kmeans algorithm using C 

#### Languages Used:
- C
- 

#### Files Added:
- K-means/K-means_sing_c.c

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by arghac14** [comment]
@asha15

File: Machine_Learning/K-means/k-means_using_c.c [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    
+    printf("\nEnter first mean: ");
+    scanf("%d", &m1);
+    printf("\nEnter second mean: ");
+    scanf("%d", &m2);
+     
+    int om1, om2;    //previous means
+    
+    do{
+
```
Comment (asha15) [inline Machine_Learning/K-means/k-means_using_c.c]: remove this line

File: Machine_Learning/K-means/k-means_using_c.c [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            t1 = -t1;
+        } 
+	    
+        t2 = a0[i] - m2;
+	
+        if(t2 < 0){
+            t2 = -t2;
+        }
+	
+        if(t1 < t2){
```
Comment (asha15) [inline Machine_Learning/K-means/k-means_using_c.c]: correct the indentation

File: Machine_Learning/K-means/k-means_using_c.c [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+            //near to first mean
+            a1[j] = a0[i];
+            j++;
+        }
+        else
+        {
+            //near to second mean
+            a2[k] = a0[i];
+            k++;
+        }
```
Comment (asha15) [inline Machine_Learning/K-means/k-means_using_c.c]: correct the indentation

**Comment by asha15** [comment]
please squash the commits before pushing it

**Comment by arghac14** [comment]
@asha15 I think now it's ready to be merged, please check.

**Comment by arghac14** [comment]
@asha15 ma'am please label the PR

**Pull request merged**